### PR TITLE
Refactor dashboard aggregation

### PR DESCRIPTION
**Pull request by emord**
This is based on https://github.com/dimagi/commcare-hq/pull/25643

Work past the first two commits:

- Introduced 3 new base classes for partition
- Reduced some inconsistencies between classes (e.g. aggregation_query vs aggregate_query)
- Added logging between each particular query
- Removed some unused methods or moved them to a child class

I think this is strictly better than the state before, and will assist in identifying the different kinds of tables in the dashboard and standardizing the processes we're following. Though let me know if you think there's a better way to organize this. I want this to come before any other work on performance improvements or splitting up more tasks.

### CODE REVIEW

**Comment by emord** [comment]
Changed the base to master so that tests can run. The first two commits are from https://github.com/dimagi/commcare-hq/pull/25643. the rest should be reviewed commit by commit anyways

File: custom/icds_reports/utils/aggregation_helpers/distributed/base.py [hunk 1]
```diff
@@ -57,11 +65,14 @@ def aggregate_query(self):
 
 class StateBasedAggregationDistributedHelper(BaseICDSAggregationDistributedHelper):
     def aggregate(self, cursor):
-        drop_query, drop_params = self.delete_old_data_query()
+        delete_query, delete_params = self.delete_old_data_query()
         agg_query, agg_params = self.aggregation_query()
 
-        cursor.execute(drop_query, drop_params)
+        logging.info(f'Deleting {self.helper_key} for {self.month} and state {self.state_id}')
```
Comment (snopoke) [inline custom/icds_reports/utils/aggregation_helpers/distributed/base.py]: These should be using `logger` and not the root logger from `logging`

**Review by snopoke — COMMENTED** [review]
This looks great!

**Comment by czue** [comment]
Point of process: was this PR approved? (I think Simon's comment could be considered ~approval though was not explicit).

I'm not complaining about it being merged, but just wondering if that was accidental or there was some offline discussion or something else...

(it was on my backlog to look but hadn't gotten there yet)

**Comment by emord** [comment]
@czue Thanks for flagging. I agree that I would not consider this PR to be approved despite Simon's positive comment. I can't say exactly what I was thinking last night, but guessing it was because I was working late and cleaning up my other PRs (this particular PR being a pre-factor), I scrolled down here, saw a resolved comment and green checks (from travis/stickler/labels), so definitely on me for merging too early.

There shouldn't have been any work that was dependent on this PR so we should be able to revert if we think that's the best way forward.

File: custom/icds_reports/utils/aggregation_helpers/distributed/base.py [hunk 2]
```diff
@@ -95,3 +95,17 @@ def delete_old_data_query(self):
...
+
+
+class StateBasedAggregationPartitionedHelper(BaseICDSAggregationDistributedHelper):
+    """Helper for tables that reside on Citus master and are partitioned into one tables per month and state
+    """
+
+    def aggregate(self, cursor):
+        drop_query = self.drop_table_query()
+        curr_month_query, curr_month_params = self.create_table_query()
+        agg_query, agg_param = self.aggregate_query()
```
Comment (czue) [inline custom/icds_reports/utils/aggregation_helpers/distributed/base.py]: nit: should this be `agg_params`?

**Review by czue — COMMENTED** [review]
lgtm

**Comment by czue** [comment]
@emord I just reviewed and you can consider it approved by me, so let's not worry about revert

**Pull request merged**